pkg/artifactory/provider/provider.go

```diff
@@ -196,7 +196,7 @@ func providerConfigure(ctx context.Context, d *schema.ResourceData, terraformVer
 
 	checkLicense := d.Get("check_license").(bool)
 	if checkLicense {
-		licenseErr := util.CheckArtifactoryLicense(restyBase)
```

Comment:
I am really starting to feel like some extension are in order. More like:

```go
  utils.InSlice(client.GetLicense(), "Enterprise","Commercial", "Edge"))
```

Reply:
Good idea. I'll write up a ticket for this 😄  In the meantime, let's get this one out first.

Reply:
With Go 1.18 one can do `slices.Contains([]string{"Enterprise", "Commercial", "Edge"}, client.GetLicense())`